### PR TITLE
Fixes the deadlock seen in papua and mpk. Tested as working Papua

### PR DESCRIPTION
**Pull request by kheimerl**

### CODE REVIEW

**Comment by kheimerl** [comment]
ptal

File: Control/L3MMLayer.cpp [hunk 1]
```diff
@@ -277,10 +277,13 @@ bool MMContext::mmCheckSipMsgs()
 	// messages during the stress test.  Ticket #1905 reports a crash that looks like using a transaction here while
 	// it is being deleted.  Since I have separated layer2 from layer3 with InterthreadQueues since the last test,
 	// I am re-enabling this global lock to attempt to fix the crash.
-	ScopedLock lock(gMMLock,__FILE__,__LINE__);	// I think this is unnecessary, but be safe.
+	// (kurtis) Update 6-17-15: This is causing deadlock; it seems that lockAndInvokeSipMsgs may not return
+	// I am disabling this lock and inserting one in that function to protect for the above
```
Comment (yosemitebandit) [inline Control/L3MMLayer.cpp]: er, are you inserting a lock somewhere?  out-of-date comment?

Comment (kheimerl) [inline Control/L3MMLayer.cpp]: Oh good point. I decided not to do that yet.

**Comment by yosemitebandit** [comment]
LGTM

**Pull request merged**